src/script/media/MediaDevicesHandler.ts

```diff
@@ -111,16 +118,27 @@ export class MediaDevicesHandler {
       if (this.availableDevices[deviceType]().length === 0) {
         return '';
       }
-      const isAvailable = this.availableDevices[deviceType]().find(
-        device =>
-          ((device as MediaDeviceInfo).deviceId || (device as ElectronDesktopCapturerSource).id) === currentDeviceId,
-      );
-      if (isAvailable) {
+      const isAvailable = this.availableDevices[deviceType]().find(device => {
+        return (
+          ((device as MediaDeviceInfo).deviceId || (device as ElectronDesktopCapturerSource).id) === currentDeviceId
+        );
+      });
+
+      const isFavorite = loadValue<String[]>(`${deviceType}-fave`)?.find(id => id === currentDeviceId);
+
+      function isMediaDevice(
+        device: MediaDeviceInfo | ElectronDesktopCapturerSource | undefined,
+      ): device is MediaDeviceInfo {
+        return (device as MediaDeviceInfo)?.deviceId !== undefined;
+      }
+
+      if (isFavorite && isAvailable) {
         return currentDeviceId;
       }
-      return MediaDevicesHandler.CONFIG.DEFAULT_DEVICE[deviceType];
+      return isMediaDevice(isAvailable)
+        ? isAvailable?.deviceId
```

Comment:
There You always have access to deviceId :)

```suggestion
        ? isAvailable.deviceId
```

Reply:
Or device can be null